src/Tools/dotnet-monitor/ServiceCollectionExtensions.cs

```diff
@@ -220,26 +220,25 @@ public static IServiceCollection ConfigureExtensions(this IServiceCollection ser
             string nextToMeFolder = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
             string progDataFolder = settings.SharedConfigDirectory;
             string settingsFolder = settings.UserConfigDirectory;
+            string dotnetToolsFolder = ToolsExtensionRepository.DotnetToolsExtensionDirectoryPath;
 
-            if (string.IsNullOrWhiteSpace(progDataFolder))
-            {
-                throw new InvalidOperationException();
-            }
-
-            if (string.IsNullOrWhiteSpace(settingsFolder))
+            if (string.IsNullOrWhiteSpace(progDataFolder)
+                || string.IsNullOrWhiteSpace(settingsFolder)
+                || string.IsNullOrWhiteSpace(dotnetToolsFolder))
             {
                 throw new InvalidOperationException();
             }
 
             // Add the folders we search to get extensions from
-            services.AddExtensionRepository(1000, nextToMeFolder);
-            services.AddExtensionRepository(2000, progDataFolder);
-            services.AddExtensionRepository(3000, settingsFolder);
+            services.AddFolderExtensionRepository(1000, nextToMeFolder);
```

Comment:
Can we remove the priority notion? We'll get them in the order in which they are registered in the service collection, which is already in the order that we want.